ndk/src/looper.rs

```diff
@@ -52,7 +52,10 @@ pub enum Poll {
     /// An event was received
     Event {
         ident: i32,
-        fd: RawFd,
+        /// # Safety
+        /// The caller should guarantee that this file descriptor remains open after it was added
+        /// via [`ForeignLooper::add_fd()`] or [`ForeignLooper::add_fd_with_callback()`].
+        fd: BorrowedFd<'fd>,
```

Comment:
I will think about this a bit more before merging (cc @rib).

This is an `fd` the user gave to us (as a `BorrowedFd`, though) so they're already required to make sure it isn't closed until it is removed from the `Looper`. Second, as written above the `unsafe` block that creates this:

```
                // SAFETY: Even though this FD at least shouldn't outlive self, a user could have
                // closed it after calling add_fd or add_fd_with_callback.
```

There's barely any relevance to `self` here; the user can totally keep using _their own_ `fd` outside the scope of `Looper`, we really have no clue what its lifetime should be.

Most importantly, let's check if `winit` can easily handle the lifetime that we've now added to `Poll<'fd>`.